src/libertem/analysis/radialfourier.py

```diff
@@ -113,30 +118,49 @@ def stack():
             radius_inner=ri,
             n_bins=n_bins,
             use_sparse=use_sparse,
-            dtype=np.complex64
+            dtype=dtype
         )
 
-        orders = np.arange(max_order + 1)
+        orders = np.arange(max_order + 1, dtype=dtype)
 
         r, phi = masks.polar_map(
             centerX=cx,
             centerY=cy,
             imageSizeX=detector_x,
             imageSizeY=detector_y
         )
-        modulator = np.exp(phi * orders[:, np.newaxis, np.newaxis] * 1j)
+        modulator = np.exp(phi.astype(dtype) * orders[:, np.newaxis, np.newaxis] * 1j)
 
         if use_sparse:
             rings = rings.reshape((rings.shape[0], 1, *rings.shape[1:]))
             ring_stack = [rings] * len(orders)
             ring_stack = sparse.concatenate(ring_stack, axis=1)
-            ring_stack *= modulator
+            _radial_mask_product(
+                ring_stack_coords=ring_stack.coords,
+                ring_stack_data_inout=ring_stack.data,
+                modulator=modulator
+            )
         else:
             ring_stack = rings[:, np.newaxis, ...] * modulator
         return ring_stack.reshape((-1, detector_y, detector_x))
     return stack
 
 
+@numba.njit(cache=True, fastmath=True, nogil=True)
```

Comment:
Is `fastmath` needed for performance? I'd rather only add it if it brings a significant speed-up; as I understand it, there are some safety concerns, for example NaN-values causing UB etc.